src/test/java/reactor/netty/http/client/HttpRedirectTest.java

```diff
@@ -583,4 +583,34 @@ public void testLastLocationSetToResourceUrlOnRedirect() throws CertificateExcep
 		initialServer.disposeNow();
 		redirectServer.disposeNow();
 	}
+
+	@Test
+	public void testBuffersForRedirectShouldBeReleased() {
+		final String initialPath = "/request";
+		final String redirectPath = "/redirect";
+
+		DisposableServer server =
+				HttpServer.create()
+						.port(0)
+						.route(r -> r
+								.get(initialPath, (req, res) -> res
+										.status(HttpResponseStatus.MOVED_PERMANENTLY)
+										.header(HttpHeaderNames.LOCATION, redirectPath)
+										.sendString(Mono.just("Redirect response content!")))
+								.get(redirectPath, (req, res) -> res.send()))
+						.wiretap(true)
+						.bindNow();
+
+		HttpClient.create()
+				.wiretap(true)
+				.followRedirect(true)
+				.get()
```

Comment:
You may try to add `ChannelInboundHandler` so that you will have access to the incoming message.
After that you can test that message with `assertThat(ReferenceCountUtil.refCnt(msg)).isEqualTo(0);`

Reply:
Good idea, thank you. I've updated the test, so it fails, when there is a leak.